src/main/kotlin/turku/forge/babyproject/api/RootApiController.kt

```diff
@@ -70,5 +70,9 @@ class RootApiController {
 @JsonInclude(Include.NON_NULL)
 class IndexResource : RepresentationModel<IndexResource>()
 
+@Relation(collectionRelation = channelRel)
 @JsonInclude(Include.NON_NULL)
-class Channel(val name: String, val subscription: String? = null) : RepresentationModel<Channel>()
+data class Channel(
+    val name: String,
+    val subscription: String? = null
```

Comment:
Does `subscription` have to be optional?

Reply:
To be honest I have no idea. But I don’t think so! Will fix that now

Reply:
Ahh just tried it in the code and since `channelLink.toUriComponentsBuilder().build().path` is nullable subscription has to be as well

Reply:
Ah okay, I guess it would be possible to add a check if `channelLink.toUriComponentsBuilder().build().path` is null then we would set it to `/channel/*name*`. But that might be wrong, I can't really say as I don't understand how everything is connected here, what do you think?